data/preprocess/preprocess_util.py

```diff
@@ -0,0 +1,46 @@
+
+def store_map(symbol_map, filename):
+    with open(filename, "w") as f:
+        for symbol, index in symbol_map.items():
+            f.write(f"{index}\t{symbol}\n")
+
+
+def process_raw_split_files(raw_split_files: dict, folder: str,  order_sop: bool = False):
+
+    if order_sop:
+        S, P, O = 0, 2, 1
+    else:
+        S, P, O = 0, 1, 2
+
+    # read data and collect entities and relations
```

Comment:
There is still a lot of code duplication. Perhaps: move to util.py things like processing a split and updating the entity/relation dictionary as well as the yaml keys.